spec/javascripts/components/story/expanded_story/expanded_story_spec.js

```diff
@@ -2,67 +2,79 @@ import React from 'react';
 import { shallow } from 'enzyme';
 import { ExpandedStory } from 'components/story/ExpandedStory/index';
 import storyFactory from '../../../support/factories/storyFactory';
+import ExpandedStoryRelease from 'components/story/ExpandedStory/ExpandedStoryRelease';
+import ExpandedStoryDefault from 'components/story/ExpandedStory/ExpandedStoryDefault';
 
 describe('<ExpandedStory />', () => {
   const defaultProps = () => ({
     story: storyFactory(),
     editStory: sinon.spy(),
     saveStory: sinon.spy(),
-    storyFailure: sinon.spy(),
-    createTask: sinon.spy(),
-    deleteTask: sinon.spy(),
-    toggleTask: sinon.spy(),
     deleteStory: sinon.spy(),
-    deleteNote: sinon.spy(),
-    createNote: sinon.spy(),
-    addLabel: sinon.spy(),
-    removeLabel: sinon.spy(),
-    setLoadingStory: sinon.spy(),
-    addAttachment: sinon.spy(),
-    removeAttachment: sinon.spy(),
-    users: [],
     project: { labels: [] },
     onToggle: sinon.spy()
   });
 
-  it('renders children components', () => {
-    const story = storyFactory({
-      _editing: {
-        ...storyFactory(),
-        _isDirty: false
-      }
+  describe('when storyType is a release', () => {
+    it('renders ExpandedStoryRelease component', () => {
+      const story = storyFactory({
+        _editing: {
+          ...storyFactory(),
+          storyType: 'release'
+        }
+      });
+
+      const wrapper = shallow(
+        <ExpandedStory
+          {...defaultProps()}
+          story={story}
+        />,
+        { disableLifecycleMethods: true }
+      );
+
+      expect(wrapper.find(ExpandedStoryRelease)).toExist();
+      expect(wrapper.find(ExpandedStoryDefault)).not.toExist();
     });
-    const project = { id: 42, labels: [] };
-    const wrapper = shallow(
-      <ExpandedStory
-        {...defaultProps()}
-        story={story}
-        project={project}
-      />,
-      { disableLifecycleMethods: true }
-    );
+  });
 
-    expect(wrapper.find('ExpandedStoryControls')).toExist();
-    expect(wrapper.find('ExpandedStoryHistoryLocation')).toExist();
+  describe("when storyType isn't a release", () => {
+    it('renders ExpandedStoryDefault component', () => {
+      const story = storyFactory({
+        _editing: {
+          ...storyFactory(),
+          storyType: 'feature'
+        }
+      });
+
+      const wrapper = shallow(
+        <ExpandedStory
+          {...defaultProps()}
+          story={story}
+        />,
+        { disableLifecycleMethods: true }
+      );
+
+      expect(wrapper.find(ExpandedStoryDefault)).toExist();
+      expect(wrapper.find(ExpandedStoryRelease)).not.toExist();
+    });
   });
 
   it('adds loading overlay when updating a story', () => {
```

Comment:
Are we checking if it really adds the overlay here? Or is it checking if it adds the overlay _class_?